common/src/main/java/dev/ftb/mods/ftblibrary/api/sidebar/SidebarButton.java

```diff
@@ -0,0 +1,42 @@
+package dev.ftb.mods.ftblibrary.api.sidebar;
+
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+
+import java.util.List;
+import java.util.function.BooleanSupplier;
+import java.util.function.Supplier;
+
+public interface SidebarButton
+{
+
+    /**
+     * @return the id of the button
+     */
+    ResourceLocation getId();
+
+    /**
+     * @param customTextHandler A supplier that returns the text to be displayed on the button
+     * @deprecated Use {@link #addOverlayRender(ButtonOverlayRender)} instead
+     */
+    @Deprecated(forRemoval = true)
```

Comment:
It shouldn't be deprecated in the commit adding it :joy:

Reply:
Not technically adding this, just moving it from the old Sidebar Button class, in order to maintain old compact, or compact in a way that stuff does not crash.